test/jit/test_freezing.py

```diff
@@ -1331,3 +1332,59 @@ def forward(self, x: torch.Tensor, key: str) -> Any:
         m.eval()
         with self.assertRaisesRegex(RuntimeError, "Freezing modules containing prim::ModuleDictIndex is not supported"):
             mf = torch._C._freeze_module(m._c)
+
+
+class TestFrozenOptimizations(JitTestCase):
+    def test_conv_bn_folding(self):
+        conv_bias = [True, False]
+        module_pairs = [(nn.Conv1d, nn.BatchNorm1d), (nn.Conv2d, nn.BatchNorm2d), (nn.Conv3d, nn.BatchNorm3d)]
+        use_tracing = [True, False]
+
+        for use_bias, modules, tracing in product(conv_bias, module_pairs, use_tracing):
+            class ConvBN(torch.nn.Module):
+                def __init__(self, in_channels, out_channels, **kwargs):
+                    super(ConvBN, self).__init__()
+                    self.conv = modules[0](in_channels, out_channels, bias=use_bias, **kwargs)
+                    self.bn = modules[1](out_channels, eps=0.001)
+
+                def forward(self, x):
+                    x = self.conv(x)
+                    return self.bn(x)
+
+            mod_eager = ConvBN(3, 32, kernel_size=3, stride=2).eval()
+            inps = [4, 3, 4]
+            if modules[0] == nn.Conv2d:
+                inps.append(inps[-1])
+            if modules[0] == nn.Conv3d:
+                inps.append(inps[-1])
+                inps.append(inps[-1])
+
+            inp = torch.rand(inps)
+
+            if tracing:
+                scripted_mod = torch.jit.trace(mod_eager, (inp))
+            else:
+                scripted_mod = torch.jit.script(mod_eager)
+
+            self.run_pass("inline", scripted_mod.graph)
+
+            # TODO: add optimization for conv size
+            # Currrently a size check in batch norm gives conv multiple uses
+            # and prevents fusion
```

Comment:
Does this mean that in practice this pass won't be able to do anything yet?

Reply:
Yes, for scripted models. I'm going to address that limitation later on in the stack